### ISSUE TITLE
[Docs]: Prefix the cnpg resource type in kubectl examples

### ISSUE THREAD

**Issue by lbruun** [cloudnative-pg/cloudnative-pg#6308]
### Is there an existing issue already for your request/idea?

- [x] I have searched for an existing issue, and could not find anything. I believe this is a new documentation enhancement to be evaluated.

### What problem in the existing documentation this issue aims to solve?

The resource type in Kubernetes, for example `cluster`, need only be unique when combined with the API group. This is a problem because several Kubernetes CRDs from various projects seems to use the name `cluster`  (and you may find some that would use the name `backup` too, etc).

If a Kubernetes cluster has several resource types with the same name, then the `kubectl get` command will list (only) the one that comes first in the retrieved API discovery document. This may be a surprising result.

To be on the safe side the resource type can be qualified when using it in a `kubectl` command.

In others words:

This may be ambiguous:

```bash
kubectl get cluster     # Show all Pg Clusters (or?)
```
.. but this is precise (and shows intent);

```bash
kubectl get cluster.postgresql.cnpg.io   # Show all Pg Clusters
```

Bottom line: I believe the qualified form should be evangelized as a kind of a best practice somewhere in the documentation .. to make sure scripts are future proof. 

### Describe what additions need to be done to the documentation

Put a comment somewhere about using _qualified_ resource type names in `kubectl`  commands. .. if you want to be safe.

### Backport?

Yes

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by pchovelon** [comment]
PR just created.

**Comment by sxd** [comment]
Looking into this, the documentation is meant to be used to be used with the cnpg operator, if you have more resources with the same name, the documentation need to assume you know this and now how to handle that because at this point, we can presume you know what you have in your cluster.

On the other hand, probably what we can do is add a shortcut in the resources and not put the full name? if we add this kind of full path in the documentation people may assume is always required and that's not the real case.

Probably there's a lot more to discuss here about how we can create a shorter and unique way to request the objects

**Comment by pchovelon** [comment]
Yes of course, we can assume that people know exactly what happens when then use `kubectl get ...`. However, depending on what is installed and who deployed it in the cluster, they might not know what is exactly inside.

Simple example : 

```
➜  /tmp k get backups
No resources found in default namespace.
➜  /tmp k get backups.postgresql.cnpg.io 
NAME                  AGE     CLUSTER      METHOD              PHASE       ERROR
backup-test    3h32m   postgresql   barmanObjectStore   completed  
```

"Just" because a new CRD has been created and that kubectl uses alphabetical order to retrieve objects.
Other tools, like Velero use Backup word.

It can be confusing. Thus, in a second thought, it might not be necessary to add all the prefixes in the documentation. Maybe a _WARNING BOX_ will be enough.

> Probably there's a lot more to discuss here about how we can create a shorter and unique way to request the objects

Yep. Sure !

CNPGCluster ? CNPGBackup ? CNPGPoolers ? 😃

**Comment by lbruun** [comment]
Another way to go about this is to launch a PR to the kubectl folks asking them to make `kubectl` barf and abort if multiple resource types match. This will then allow users to continue to use the lazy syntax (the short one) without the current risk.

Here is the PR:   https://github.com/kubernetes/kubectl/issues/1704

This is not to say that the current PR about a "something" in the cnpg docs about this topic isn't justified too. It is.
